common/range-map/src/range_wrapper.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::cmp::Ordering;
+use core::ops::Range;
+
+#[derive(Eq, Debug, Clone)]
+pub struct RangeWrapper<T> {
+    pub range: Range<T>,
+}
+
+impl<T> RangeWrapper<T> {
+    pub fn new(range: Range<T>) -> RangeWrapper<T> {
+        RangeWrapper { range }
+    }
+}
+
+impl<T> PartialEq for RangeWrapper<T>
+where T: Eq
+{
+    fn eq(&self, other: &RangeWrapper<T>) -> bool {
+        self.range.start == other.range.start && self.range.end == other.range.end
+    }
+}
+
+impl<T> Ord for RangeWrapper<T>
+where T: Ord + std::fmt::Debug
```

Comment:
To manually implement a total order value, these properties have to be satisfied or a data structure such as btree or binary search won't work as expected:

https://en.wikipedia.org/wiki/Total_order

![image](https://user-images.githubusercontent.com/44069/161027771-e4b4d236-aefc-476e-be1d-31460d61450e.png)

Thus tests that ensure these properties will be needed.